### ISSUE TITLE
Android logs "[unexpected] duplicate Hostinfo" once a second

### ISSUE THREAD

**Issue by bradfitz** [tailscale/tailscale#611]
In `control/controlclient/auto.go`, this `[unexpected]` log line is firing once a second on Android (at 77cad13c7083039).

```
func (c *Client) SetHostinfo(hi *tailcfg.Hostinfo) {
        if hi == nil {
                panic("nil Hostinfo")
        }
        if !c.direct.SetHostinfo(hi) {
                c.logf("[unexpected] duplicate Hostinfo: %v", hi)
                return
        }
```

/cc @dshynkev

**Comment by dshynkev** [comment]
Also observing this on Linux at HEAD. Trying to understand the related state machinery is part of the reason why I haven't merged #609 yet. That line has existed for a while, but it's not quite clear to me why, since duplicate `Hostinfo`s seem like a perfectly natural thing to have.

**Comment by bradfitz** [comment]
I think prior to ShieldsUp, cli.SetHostinfo was only called on changes, and I added that unexpected log because that was actually true for a bit. But then `ipn/local.go` changed with `LocalBackend.doSetHostinfoFilterServices` etc and that wasn't true anymore.

That code could be cleaned up and simplified probably. I see a TODO from @danderson about that.
